src/NUnitFramework/framework/Internal/Execution/SimpleWorkItemDispatcher.cs

```diff
@@ -66,34 +66,35 @@ public void Dispatch(WorkItem work)
                 _topLevelWorkItem = work;
                 _runnerThread = new Thread(RunnerThreadProc);
                 _runnerThread.Start();
-            }
+			}	
 #endif
         }
-        
-#if !PORTABLE
-        private void RunnerThreadProc()
-        {
-            _topLevelWorkItem.Execute();
-        }
-#endif
 
-        /// <summary>
-        /// Cancel the ongoing run completely.
-        /// If no run is in process, the call has no effect.
-        /// </summary>
-        public void CancelRun()
-        {
 #if !PORTABLE
+    private void RunnerThreadProc()
+    {
 #if NETCF
-            // NETCF: Check if this can be done better
-            if (_runnerThread != null)
-                _runnerThread.Abort();
-#else
-            if (_runnerThread != null && _runnerThread.IsAlive)
-                ThreadUtility.Kill(_runnerThread);
+        _topLevelWorkItem.Context.EstablishExecutionEnvironment();
```

Comment:
But this line is only executed when we start the top level test. Reseting it from one test to another would be the job of CompositeWorkItem in the non-parallel environment.